### PR TITLE
Prep for 2017-05-17 EOY emails.

### PR DESCRIPTION
**Pull request by ashercodeorg**
The plan is to send the `k5nopd` experiment emails (templates `k5nopdA` and `k5nopdB`) to determine an A/B test winner. After updating all six templates appropriately, the main campaign emails will be sent.

UPDATE: After sending the initial experiment emails, it was realized that the messages did not contain tracking pixels. Thus an additional 18K emails were removed from the remaining file and divided between experiment-3 and experiment-4.

As discussed offline, note that `pegasus.contact_rollups.roles` includes `Teacher` for Code Studio accounts with `user_type` of `teacher` and for certain `pegasus.forms` contacts. Both groups are intended recipients of this email campaign.

### CODE REVIEW

File: bin/mail/2017-05-17-end-of-year/generate-list [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    AND roles LIKE '%Teacher%'
+    AND (country = 'United States' OR country IS NULL)
+    AND (
+      (
+        grades_taught LIKE '%K%'
+          OR grades_taught LIKE '%,1,%' OR grades_taught = '1' OR grades_taught LIKE '1,%' OR grades_taught LIKE '%,1'
+          OR grades_taught LIKE '%,2,%' OR grades_taught = '2' OR grades_taught LIKE '2,%' OR grades_taught LIKE '%,2'
+          OR grades_taught LIKE '%,3,%' OR grades_taught = '3' OR grades_taught LIKE '3,%' OR grades_taught LIKE '%,3'
+          OR grades_taught LIKE '%,4,%' OR grades_taught = '4' OR grades_taught LIKE '4,%' OR grades_taught LIKE '%,4'
+          OR grades_taught LIKE '%,5,%' OR grades_taught = '5' OR grades_taught LIKE '5,%' OR grades_taught LIKE '%,5'
```
Comment (aoby) [inline bin/mail/2017-05-17-end-of-year/generate-list]: Wow, this is cumbersome. Can we use [FIND_IN_SET](https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_find-in-set) instead?

```
FIND_IN_SET('1', grades_taught) > 0 
OR FIND_IN_SET('2', grades_taught) > 0 
# etc
```

File: bin/mail/2017-05-17-end-of-year/generate-list [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+      ) OR (
+        ages_taught LIKE '%,4,%' OR ages_taught = '4' OR ages_taught LIKE '4,%' OR ages_taught LIKE '%,4'
+          OR ages_taught LIKE '%,5,%' OR ages_taught = '5' OR ages_taught LIKE '5,%' OR ages_taught LIKE '%,5'
+          OR ages_taught LIKE '%,6,%' OR ages_taught = '6' OR ages_taught LIKE '6,%' OR ages_taught LIKE '%,6'
+          OR ages_taught LIKE '%,7,%' OR ages_taught = '7' OR ages_taught LIKE '7,%' OR ages_taught LIKE '%,7'
+          OR ages_taught LIKE '%,8,%' OR ages_taught = '8' OR ages_taught LIKE '8,%' OR ages_taught LIKE '%,8'
+          OR ages_taught LIKE '%,9,%' OR ages_taught = '9' OR ages_taught LIKE '9,%' OR ages_taught LIKE '%,9'
+          OR ages_taught LIKE '%,10,%' OR ages_taught = '10' OR ages_taught LIKE '10,%' OR ages_taught LIKE '%,10'
+          OR ages_taught LIKE '%,11,%' OR ages_taught = '11' OR ages_taught LIKE '11,%' OR ages_taught LIKE '%,11'
```
Comment (aoby) [inline bin/mail/2017-05-17-end-of-year/generate-list]: `FIND_IN_SET` here too, and other similar queries below

File: bin/mail/2017-05-17-end-of-year/send-final-real-mails [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+../send-to-mailing-list 2017-05-17-endofyear-k5pd ./email_1_csf_pd.csv
+../send-to-mailing-list 2017-05-17-endofyear-k5nopd ./email_2_csf_no_pd-remainder.csv
```
Comment (aoby) [inline bin/mail/2017-05-17-end-of-year/send-final-real-mails:4]: Do you want to send the remainder from the A/B tests in this initial barrage? I think you want to send this separately sometime later, after determining the winner.

Comment (ashercodeorg) [reply bin/mail/2017-05-17-end-of-year/send-final-real-mails:4]: My belief is yes, though I'm confused by your question. My intent was to send `send-experiment-real-emails` to execute the A/B test and determine a winner, update the templates appropriately (the winner will impact all six email templates), then send `send-final-real-emails`. Am I missing something obvious?

Comment (aoby) [reply bin/mail/2017-05-17-end-of-year/send-final-real-mails:4]: Oh, I thought the winner was only relevant to that one template and we were sending the other 5 at the same time as the A/B test for `k5nopd`, then the winner later. In that case, this is fine.

Comment (ashercodeorg) [reply bin/mail/2017-05-17-end-of-year/send-final-real-mails:4]: Sorry, I should have made that clear somewhere. Added a comment in the PR summary.

**Comment by jeremydstone** [comment]
LGTM as well

**Comment by ashercodeorg** [comment]
As this needs to exist on `production-daemon` immediately (to send the experiment emails), so as to avoid merge conflicts, my intent is to merge this PR only after all associated emails have been sent.

**Comment by ashercodeorg** [comment]
As these are now being sent Monday AM, I am going to merge this PR in hopes of it reaching production-daemon before then. Note that I have the existing files backed up on production-daemon and will do a diff to confirm that the old and new are identical before running anything.

**Pull request merged**